### DISCUSSION TITLE
GitHub action: deploying a new site when someone creates a pull request

### DISCUSSION THREAD

**Discussion by Fullchee** [squidfunk/mkdocs-material#4800] [General]
When I submit a PR for review, the reviewer can use the GitHub rich diff.

<img width="350" alt="image" src="https://user-images.githubusercontent.com/11246258/209450865-02b76280-1c2f-409f-b9dd-fc45133cafe1.png">

However, they can't see MKDocs Material specific features like admonitions and icons.

Does anyone have any experience creating a GitHub action that will launch a GitHub pages site when you create a new pull request?

so that the reviewer gets a better reviewing experience in the browser?

## Can't the reviewer just pull the changes and view it locally?

I've made the setup process as simple as possible

1. `pip install pipenv`
2. `pipenv run start`

but the few people who have updated the docs mostly do so in the browser on GitHub.com

## Background

I use GitHub Pages to host my company's MKDocs site

- so that only people in the org can access the site

**Comment by squidfunk** [comment]
You can use [Netlify deploy previews](https://docs.netlify.com/site-deploys/deploy-previews/) for this. However, thinking about it, it might be possible to craft up a process with [mike](https://github.com/jimporter/mike).

**Reply by joapuiib** [reply]
Has anybody been able to pull it out using [mike](https://github.com/jimporter/mike)?
